### PR TITLE
(chore): Initial project setup.

### PR DESCRIPTION
**Pull request by alxhub**
NPM, Typescript, and Gulp configurations.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/node_modules
+/spec
```
Comment (jeffbcross) [inline .gitignore]: What is in spec?

Comment (alxhub) [inline .gitignore]: Hmm, that might be left over from when I was trying to get Jasmine to work.

Comment (alxhub) [inline .gitignore]: Fixed.

**Comment by jeffbcross** [comment]
LGTM!

**Pull request merged**